analyzers/src/SonarAnalyzer.Common/Helpers/SonarAnalysisContext.cs

```diff
@@ -90,6 +91,12 @@ internal SonarAnalysisContext(AnalysisContext context, IEnumerable<DiagnosticDes
         public bool ShouldAnalyzeGenerated(Compilation c, AnalyzerOptions options) =>
             ShouldAnalyzeGenerated(context, c, options);
 
+        public bool IsUnchanged(SyntaxTree syntaxTree, Compilation compilation, AnalyzerOptions options) =>
+            UnchangedFilesCache.GetValue(compilation, _ => CreateUnchangedFilesHashSet(options)).Contains(syntaxTree.FilePath);
+
+        private ImmutableHashSet<string> CreateUnchangedFilesHashSet(AnalyzerOptions options) =>
+            ImmutableHashSet.Create(StringComparer.OrdinalIgnoreCase, ProjectConfiguration(options).AnalysisConfig.UnchangedFiles());
```

Comment:
After the contract change, this will need to be like this.
```
        private static ImmutableHashSet<string> CreateUnchangedFilesHashSet(TryGetValueDelegate<ProjectConfigReader> tryGetValue, AnalyzerOptions options) =>
            ImmutableHashSet.Create(StringComparer.OrdinalIgnoreCase, ProjectConfiguration(tryGetValue, options).AnalysisConfig.UnchangedFiles());
```